src/contexts/UserState.tsx

```diff
@@ -91,24 +97,26 @@ export function UserProvider({ children }: UserProviderProps) {
     window.location.href = '/login'
   }
 
-  const onLogin = (
-    JWT: string,
-    user: User | undefined = undefined,
-    permissions: TemplatePermissions | undefined = undefined
-  ) => {
+  const onLogin: OnLogin = (JWT: string, user, permissions, orgList, isAdmin) => {
     if (JWT == undefined) logout()
     dispatch({ type: 'setLoading', isLoading: true })
     localStorage.setItem('persistJWT', JWT)
     if (!user || !permissions) fetchUserInfo({ dispatch: setUserState }, logout)
     else {
-      dispatch({ type: 'setCurrentUser', newUser: user, newPermissions: permissions })
+      dispatch({
+        type: 'setCurrentUser',
+        newUser: user,
+        newPermissions: permissions || {},
+        newOrgList: orgList || [],
+        newIsAdmin: !!isAdmin,
+      })
       dispatch({ type: 'setLoading', isLoading: false })
     }
   }
 
   // Initial check for persisted user in local storage
   const JWT = localStorage.getItem('persistJWT')
-  if (JWT == 'undefined') logout()
+  if (JWT == undefined) logout()
```

Comment:
Yes indeed -- this needs to be changed back to fix the perpetual loop I mentioned in other comment!

Reply:
d140f22a8e2bbdbd6c64ac045290d1b1942610e8